src/Components/Generator/DynamicInputContainer.tsx

```diff
@@ -0,0 +1,189 @@
+import type { ComponentChildren } from 'preact';
+import { useMemo } from 'preact/hooks';
+import { DynamicInput } from './DynamicInput';
+import { Text, MarkupText, IntlProvider } from 'preact-i18n';
+import t from '../../Utility/i18n';
+import { EMTPY_ADDRESS, IdDataElement } from 'request';
+import { useState } from 'preact/hooks';
+import { isAddress } from '../../Utility/requests';
+
+type DynamicInputContainerProps = {
+    id: string;
+    title: string;
+    headingClass: string;
+    fields: IdDataElement[];
+    fillFields?: IdDataElement[];
+
+    onAddField: (field: IdDataElement) => void;
+    onRemoveField: (id: number) => void;
+    onChange: (id: number, field: IdDataElement) => void;
+
+    allowAddingFields?: boolean;
+    allowRemovingFields?: boolean;
+    allowChangingFieldDescriptions?: boolean;
+    hasPrimary?: boolean;
+
+    children: ComponentChildren;
+};
+
+export const DynamicInputContainer = (props: DynamicInputContainerProps) => {
+    const [inputTypeToAdd, setInputTypeToAdd] = useState<IdDataElement['type']>('input');
+
+    const address_number = useMemo(
+        () => props.fields.reduce((total, field) => total + (field.type === 'address' ? 1 : 0), 0),
+        [props.fields]
+    );
+    const input_elements = props.fields.map((field, i) => {
+        const field_uid = field.type + i; // TODO: This is not ok. It needs to be proper unique id! cf. #880 and #881
+        return (
+            <DynamicInput
+                key={field_uid}
+                id={field_uid}
+                suffix={props.id}
+                optional={field.optional}
+                onChange={(field) => props.onChange(i, field)}
+                onRemove={() => {
+                    if (isFieldEmpty(props.fields[i]) || window.confirm(t('confirm-input-remove', 'generator'))) {
+                        props.onRemoveField(i);
+                    }
+                }}
+                hasPrimary={props.hasPrimary && address_number > 1}
+                value={field}
+                allowRemoving={props.allowRemovingFields ?? true}
+                allowChangingDescription={props.allowChangingFieldDescriptions ?? true}
+            />
+        );
+    });
+
+    const addFillField = (newField: IdDataElement) => {
+        props.fields.forEach((field, key) => {
+            if (['name', 'birthdate', 'email'].includes(field.type) && field.type === newField.type) {
+                newField.desc = field.desc;
```

Comment:
True. This is also not what happend before: We just replaced the value of the email field is the description changed (even though the value might not have), sot it was extremely unintuitive. I decided to change that now, so that we add fields as a new one if the description changes but not the value, but we replace the value if only that changed.